### ISSUE TITLE
Broken dependencies in kali and debian templates

### ISSUE THREAD

**Issue by Middenvaus** [QubesOS/qubes-issues#8434]
[How to file a helpful issue](https://www.qubes-os.org/doc/issue-tracking/)

### Qubes OS release
4.1

### Brief summary

I can't install anything adequately via apt due to broken dependencies in the above templates. If I try to install something, then apt constantly tries to remove the necessary packages from the qubes-core list. If you try to freeze them, you will not be able to install any package through apt, as it wants to remove packages from the qubes-core list

#### In terminal:
apt upgrade

### Temporary solution
You can live with this if you use an add-on for apt aptitude. You can use it to install packages, with the exception of some, and even update. This will work on kali and debian template.

#### In terminal
For install packages:
aptitude install <package_name>

For safe upgrade without deleting necessary packages in dependencies:
aptitude safe-upgrade

**Comment by andrewdavidwong** [comment]
I and many others install stuff in Debian templates on Qubes 4.1 all the time with no problems. Which Debian template are you using? Is there anything unusual about your setup?

**Comment by Middenvaus** [comment]
I am currently using the debian 11 template with Linux kernel debian-11

I had this problem for a long time and I wrote about it just now. I will check again

**Comment by Middenvaus** [comment]
I ran diagnostics and found the problem. Here is apt output when trying to set xfce4-settings in kali template:
![2023-08-18-190234](https://github.com/QubesOS/qubes-issues/assets/142319829/66f41bdb-b434-4a57-8140-77104595cc38)

**Comment by marmarek** [comment]
Does this happen in plain debian-11 (not kali) template too?

**Comment by Middenvaus** [comment]
> Does this happen in plain debian-11 (not kali) template too?

no, I only have a problem with this package in the kali template, but I had the same problem with other packages in the debian template

**Comment by andrewdavidwong** [comment]
(@Middenvaus, future tip: It's better to post text than images, as some people participate via plain-text email and won't be able to see your images. This applies doubly when it's just an image of plain text in the first place, as in this case.)

**Comment by unman** [comment]
@Middenvaus I have no idea what specific issues you are having, because
(as Andrew explained) I cannot see the image you posted.

You dont say what Kali template you are using, or where you got it
from.
Kali is based on Debian testing - it's inevitable that things will
break, particularly when interacting with Qubes packages.
Take a look at https://qubes.3isec.org/Templates_4.1/ where I give some
advice on how to handle Qubes packages.

**Comment by Middenvaus** [comment]
I do not urgently need to have my problem solved. I wrote here to at least know about the problem. I will write here as I use the template and find new errors or solutions to old problems in it.

**Comment by unman** [comment]
I do not see anything here worth keeping open.
If this issue is to be kept open, then the title should be changed
(removal of "Debian"), and full detail given - 
Where did Kali template come from?
What is underlying Debian release? 
Has system been release upgraded?
What packages are throwing error?
etc etc

Without this level of detail, there is no point in keeping open.
At the moment there is no clarity as to what the problem is, and if it
is something that Qubes can solve.

**Comment by andrewdavidwong** [comment]
Closing as "cannot reproduce" (we were unable to reproduce this issue). If anyone believes this is a mistake, or if anyone can reproduce the issue, please leave a comment, and we'll be happy to reopen this. Thank you.

**Comment by Middenvaus** [comment]
> @Middenvaus I have no idea what specific issues you are having, because (as Andrew explained) I cannot see the image you posted. You dont say what Kali template you are using, or where you got it from. Kali is based on Debian testing - it's inevitable that things will break, particularly when interacting with Qubes packages. Take a look at https://qubes.3isec.org/Templates_4.1/ where I give some advice on how to handle Qubes packages.
> […](#)

After install kali template from this url my problem with dependencies has been solved.

**Comment by unman** [comment]
Good - I should probably have that template updated to Trixie.

**Comment by anomaliesintent112** [comment]
I can add to this thread for clarification. I have fresh installed the Kali-core template multiple times today always with the same result. after adding the Kali repository I run sudo apt update and then sudo apt upgrade with no issue but the moment you try to install anything for example kali-linux-everything it will spit out the usual message with a section about packages to be removed. If you accept the install it bricks the template and you get the standard qrexec failed to connect to agent within 60 seconds loop of death when you try to start the Template. 

The following packages will be REMOVED:
  libatk-bridge2.0-0 libatk1.0-0 libatspi2.0-0 libcups2 libglib2.0-0
  libglib2.0-bin libgnutls30 libgtk-3-0 libhogweed6 libnettle8 libpng16-16
  mate-notification-daemon qubes-core-agent-network-manager
  qubes-vm-recommended
 

dom0 uname: Linux dom0 6.6.77-1.qubes.fc37.x86_64 #1 SMP PREEMPT_DYNAMIC Mon Feb 24 09:30:33 GMT 2025 x86_64 x86_64 x86_64 GNU/Linux

Kali-core uname: Linux kali-core 6.6.77-1.qubes.fc37.x86_64 #1 SMP PREEMPT_DYNAMIC Mon Feb 24 09:30:33 GMT 2025 x86_64 GNU/Linux

I am working on a workaround but for now the template is unusable. Please let me know if you need more details or any additional information.

**Comment by anomaliesintent112** [comment]
After much testing I have found a workaround but the Kali-core template still remains in a broken and unusable state. You can create a Qube as a standalone VM and after its created launch the settings make sure the mode is set to HVM and then boot from the Kali ISO. it may take a bit but you should have a fully functional Kali machine to work on. It sadly doesn't support the other features of Qubes but aside from that it works just fine.

**Comment by marmarek** [comment]
Trixie should work now. Users of old kali templates will need to switch to trixie repositories (from bookworm), but fresh template build (not done yet) should have that included already.

**Comment by shaaati** [comment]
I understand that this might not be a priority and that Kali is a community template. But is there any estimation on when/if we can expect a new template build for a (working) trixie-based kali-core?
